packages/arb-token-bridge-ui/src/components/TransferPanel/AdvancedSettings.tsx

```diff
@@ -15,21 +40,46 @@ export const AdvancedSettings = ({
 }: {
   destinationAddress?: string
   onChange: (value?: string) => void
-  error: AdvancedSettingsErrors | null
+  error: DestinationAddressErrors | null
 }) => {
   const {
-    app: { selectedToken }
+    app: { selectedToken, isDepositMode }
   } = useAppState()
+  const { l1, l2 } = useNetworksAndSigners()
   const { isEOA = false, isSmartContractWallet = false } = useAccountType()
 
   const [collapsed, setCollapsed] = useState(true)
+  const [warning, setWarning] = useState<string | null>(null)
 
   useEffect(
     // Show on page load if SC wallet since destination address mandatory
     () => setCollapsed(!isSmartContractWallet),
     [isSmartContractWallet]
   )
 
+  useEffect(() => {
+    // checks if trying to send to a contract address
+    async function getWarning() {
+      // only check for EOA, contract wallets will often send to another contract wallet
+      if (isEOA && isAddress(destinationAddress)) {
+        const isContractAddress = await addressIsSmartContract(
+          destinationAddress,
+          isDepositMode ? l2.provider : l1.provider
+        )
+        setWarning(
+          isContractAddress
```

Comment:
as discussed in standup, we'd like to let EOA send to an SCW because for example, users of EOA on L1 might want to send funds to their SCW on L2